src/helpers/sorts.ts

```diff
@@ -57,3 +69,86 @@ export function getDefaultTreemapSort(afm: AFM.IAfm, resultSpec: AFM.IResultSpec
 
     return [];
 }
+
+export function sortStackedSeriesData(
+    series: ISeriesItem[],
+    categories: ICategory[],
+    isSingleAxis: boolean,
+): void {
+    const isPrimarySeries = (ser: ISeriesItem) => ser.yAxis === 0;
+    const categoriesSize = categories.length;
+    const filteredSeries = isSingleAxis ? series : series.filter(isPrimarySeries);
+    const seriesData = filteredSeries.map((ser: ISeriesItem) => ser.data);
+    const seriesNo = seriesData.length;
+    const seriesDataSize = seriesNo ? seriesData[0].length : 0;
+    const subCategoriesNo = seriesDataSize / categoriesSize; // 1 mean that have no group of categories
+
+    // don't need to sort
+    if (categoriesSize === 0 || seriesDataSize === 0) {
+        return;
+    }
+
+    const indexSortOrder: ISortingData[] = buildIndexSortOrder(seriesData, categories, subCategoriesNo);
+    sortData(series, subCategoriesNo, indexSortOrder);
```

Comment:
return the new `series` object is better than override its property 
what do you think?

Reply:
I cannot agree more.
We should return new series instead of modifying themself.

Reply:
i like to keep reference to series and categories. it look cooler - don't need re-assign the return values. But anyway, i will update go back to traditional way :D